pages/_middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { NextMiddlewareResult } from "next/dist/server/web/types";
+import { NextRequest, NextResponse } from "next/server";
+
+import { GEOLOCATION } from "@/lib/const";
+import { CHANNELS, DEFAULT_CHANNEL, DEFAULT_LOCALE, LOCALES } from "@/lib/regions";
+
+export function LocaleRedirectionMiddleware({
+  nextUrl,
+  headers,
+  geo,
+}: NextRequest): NextMiddlewareResult | Promise<NextMiddlewareResult> {
+  if (nextUrl.pathname !== "/") {
+    // redirect should only be applied on homepage, without any region/locale chosen
```

Comment:
I think this is too narrow, it should catch all routes that don't include region/locale. If you remove locale/region from products pages it will throw an error

Reply:
By error you mean 404? That's expected. 
I would like to have geolocation only for the homepage. If the user visits the storefront from page URL IMO we should **not** redirect automatically. 
If there's a difference between URL and geolocation we could show a notification, but that's outside of the scope of this PR.